LRDOCS-3683 SAML Docs Update

sez11a opened the pull request:
Here's my draft of the updates. Please review for accuracy; thanks!

mwilliams2014 committed:
LRDOCS-3564 Added note on staging support to intro.

chrislui committed:
LRDOCS-3786 06-publishing-rss-feeds.markdown

Added note about needing Java proxy properties when you are behind a proxy.

mwilliams2014 committed:
LRDOCS-3748 Reorganized and wordsmithed Lexicon tutorial

mwilliams2014 committed:
LRDOCS-3748 Reorganized and wordsmithed Lexicon tutorial

mwilliams2014 committed:
LRDOCS-3748 wordsmithing

mwilliams2014 committed:
LRDOCS-3748 wordsmithing

mwilliams2014 committed:
LRDOCS-3748 wordsmithing

mwilliams2014 committed:
LRDOCS-3748 wordsmithing

blzaugg committed:
LRDOCS-3748 Distinguish Lexicon from Lexicon Experience Language.

blzaugg committed:
LRDOCS-3748 Wordsmithing

blzaugg committed:
LRDOCS-3748 Use https://liferay.github.io/clay/ for all Lexicon site links.

mwilliams2014 committed:
Merge branch 'pull-request-9-LRDOCS-3748' of https://github.com/blzaugg/liferay-docs into LRDOCS-3748-lexicon-tut-refactor

# Conflicts:
#	develop/tutorials/articles/110-portlets/20-lexicon/00-lexicon-intro.markdown
#	develop/tutorials/articles/110-portlets/20-lexicon/02-applying-lexicon-patterns-to-your-forms-navigation-and-search.markdown
#	develop/tutorials/articles/110-portlets/20-lexicon/04-adding-the-management-bar/01-implementing-the-management-bar-display-styles.markdown
#	develop/tutorials/articles/110-portlets/20-lexicon/07-using-lexicon-icons-in-your-app.markdown

jrao committed:
LRDOCS-3683 Update SAML docs as per ticket and the new SAML Admin UI updates

rbohl committed:
LRDOCS-3977 edited introduction

rbohl committed:
LRDOCS-3977 edited file naming information

rbohl committed:
LRDOCS-3977 Updates: synced with google doc

rbohl committed:
LRDOCS-3977 removed rogue figure code

rbohl committed:
LRDOCS-3817 restore workflow articles to Learning Path folder

sez11a committed:
Merge branch 'LRDOCS-3683' of github.com:jrao/liferay-docs into LRDOCS-3683-SAML-update

rbohl committed:
LRDOCS-3977 moved Typed Values section

rbohl committed:
LRDOCS-3977 reorganized Type Values section

rbohl committed:
LRDOCS-3977 explain multi-value settings

rbohl committed:
LRDOCS-3977 remove link to sling documentation

rbohl committed:
LDOCS-3817 editing intro

rbohl committed:
LRDOCS-3817 reorg: update the service layer first

rbohl committed:
LRDOCS-3817 update guestbook service to support status fields

rbohl committed:
LRDOCS-3817 service layer work

rbohl committed:
LRDOCS-3817 reordering, renaming

rbohl committed:
LRDOCS-3817 changing article titles

rbohl committed:
LRDOCS-3817 renamed folder

rbohl committed:
LRDOCS-3817 delete unnecessary article

rbohl committed:
LRDOCS-3817 merge master

sez11a committed:
Merge branch 'master' into LRDOCS-3683-SAML-update

sez11a committed:
LRDOCS-3683 Initial edits/definitions.

rbohl committed:
committing current code for review

rbohl committed:
LRDOCS-3817 workflow handler for guestbooks

rbohl committed:
LRDOCS-3817 guestbook workflow hander article, complete draft

rbohl committed:
LRDOCS-3817 entry workflow handlers article, complete draft

rbohl committed:
LRDOCS-3817 reorganized workflow handlers section

rbohl committed:
LRDOCS-3817 added article on creating new finders/getters

rbohl committed:
LRDOCS-3817 removed old handler articles

rbohl committed:
LRDOCS-3817 intro for updating the UI section

rbohl committed:
LRDOCS-3817 editing workflow intro

rbohl committed:
LRDOCS-3817 edit workflowing service layer intro

rbohl committed:
LRDOCS-3817 edited updating guestbook service layer

rbohl committed:
LRDOCS-3817 editing entry service layer updates

rbohl committed:
LRDOCS-3817 edited finding entities by status article

rbohl committed:
LRDOCS-3817 editing intro to workflow handlers section

rbohl committed:
LRDOCS-3817 editing workflow handlers articles

rbohl committed:
LRDOCS-3817 Edits to updating the UI articles, including images

rbohl committed:
LRDOCS-3817 editing and deleting unnecessary files

rbohl committed:
LRDOCS-3817 deleting old articles

sez11a committed:
Merge branch 'master' into LRDOCS-3683-SAML-update

sez11a committed:
LRDOCS-3683 Broke up tutorials.

sez11a committed:
LRDOCS-3683 Last definition.

sez11a committed:
LRDOCS-3683 Added configuration section.

rbohl committed:
LRDOCS-3817 cleanly merge master, overwrite my code changes

rbohl committed:
LRDOCS-3817 russ-QA: editing workflow intro

rbohl committed:
LRDOCS-3817 russ-QA edits to workflow-enabling the service layer

sez11a committed:
Change build.properties to reflect new branch

sez11a committed:
Header IDs

sez11a committed:
metadata file should never have been in the repo

rbohl committed:
LRDOCS-3817 russ-QA, handler articles and images

rbohl committed:
LRDOCS-3817 russ-QA edits to view layer articles

rbohl committed:
LRDOCS-3817 russ-QA: adding remaining workflow code updates

rbohl committed:
LRDOCS-3817 russ-QA: some cleanup

rbohl committed:
LRDOCS-3977 merge master

codyhoag committed:
Weekly Gradle plugin ref docs

rbohl committed:
LRDOCS-3817 added language property needed for assets and workflow

rbohl committed:
LRDOCS-3817 insert directive to add language keys

rbohl committed:
LRDOCS-3977 fix typo

stian-sigvartsen commented on discover/deployment/articles-dxp/03-configuring-liferay/04-securing-liferay/01-saml/01-setting-up-identity-provider.markdown:
```diff
@@ -0,0 +1,271 @@
+produced is signed.
+
+**SSL Required:** When this box is checked, any SAML messages that are *not*
+sent over SSL are rejected. This does not affect how URLs are generated.
```
URLs in the generated metadata are affected. Which means the peer will pick that up and send messages with HTTPS.

stian-sigvartsen commented on discover/deployment/articles-dxp/03-configuring-liferay/04-securing-liferay/01-saml/01-setting-up-identity-provider.markdown:
```diff
@@ -0,0 +1,271 @@
+
+The New Service Provider page includes these options:
+
+**Name:** The name of the Service Provider with which to connect.
```
Maybe make it clear that this name is only what you would like to call the SP on the IdP portal instance. i.e. It is purely cosmetic, but will be displayed in page content during SLO.

stian-sigvartsen commented on discover/deployment/articles-dxp/03-configuring-liferay/04-securing-liferay/01-saml/01-setting-up-identity-provider.markdown:
```diff
@@ -0,0 +1,271 @@
+file or you can manually upload the Service Provider metadata XML file. If you
+provide a URL, the XML file is automatically retrieved and periodically
+polled for updates. The update interval can be configured in
+`portlet.properties` with the `saml.metadata.refresh.interval` property which
```
saml.metadata.refresh.interval should be set via System Settings ("SAML Configuration") or via config file named com.liferay.saml.runtime.configuration.SamlConfiguration.cfg

stian-sigvartsen commented on discover/deployment/articles-dxp/03-configuring-liferay/04-securing-liferay/01-saml/01-setting-up-identity-provider.markdown:
```diff
@@ -0,0 +1,271 @@
+**Keep Alive URL:** If users are logged into several @product@ SP instances via
+a @product@ IdP, their sessions can be kept alive as long as they keep a
+browser window open to one of them. Configure this only if the SP is @product@.
+The URL is `https://[host name]/c/portal/saml/keep_alive`. 
```
1) I think this paragraph was intended to be after the next one
2) Maybe replace [host name] with [SP host name] to make it clearer that it's the peer's URL which should be provided

stian-sigvartsen commented on discover/deployment/articles-dxp/03-configuring-liferay/04-securing-liferay/01-saml/01-setting-up-identity-provider.markdown:
```diff
@@ -0,0 +1,271 @@
+
+	Note that the SAML keystore is created by the SAML app's keystore
+	manager. As described above, this keystore has two storage options, file
+	system storage (the default) and documents and media storage. 
```
Maybe mention that both a keystore and certificate + private key entry is created.
Also, the keystore is only created if it does not already exist, otherwise the existing one is used to store the new certificate + private key entry, so long as @Product is configured with the correct "Key Store password" in Control Panel → System Settings → SAML Configuration.

sez11a replied:
I was assuming that readers would know the difference between the keystore and what it contains, but maybe that was a bad assumption: I can clarify.

stian-sigvartsen replied:
That might be a reasonable assumption. I'm thinking that it by separating our the concepts it becomes easier later to talk about scenarios such as adding a CA issued certificate to a generated keystore using the offline keytool

stian-sigvartsen commented on discover/deployment/articles-dxp/03-configuring-liferay/04-securing-liferay/01-saml/02-setting-up-service-provider.markdown:
```diff
@@ -0,0 +1,279 @@
+
+**Keep Alive URL:** If users are logged into several @product@ SP instances via
+a @product@ IdP, their sessions can be kept alive as long as they keep a
+browser window open to one of them. Configure this only if the SP is @product@.
```
Replace "SP" with "IdP"

stian-sigvartsen commented on discover/deployment/articles-dxp/03-configuring-liferay/04-securing-liferay/01-saml/02-setting-up-service-provider.markdown:
```diff
@@ -0,0 +1,279 @@
+a @product@ IdP, their sessions can be kept alive as long as they keep a
+browser window open to one of them. Configure this only if the SP is @product@.
+The URL is `https://[host name]/c/portal/saml/keep_alive`. On the @product@ SP,
+configure this URL the same way, but point back to this IdP. 
```
Same again, replace "SP" with "IdP" and vica-versa

stian-sigvartsen commented on discover/deployment/articles-dxp/03-configuring-liferay/04-securing-liferay/01-saml/02-setting-up-service-provider.markdown:
```diff
@@ -0,0 +1,279 @@
+XML file or you can manually upload it. If you provide a URL, the XML file is
+automatically retrieved and periodically polled for updates. You can change the
+update interval in `portlet.properties` by modifying the
+`saml.metadata.refresh.interval` property which specifies a number of seconds.
```
As before: saml.metadata.refresh.interval should be set via System Settings ("SAML Configuration") or via config file named com.liferay.saml.runtime.configuration.SamlConfiguration.cfg

sez11a replied:
I'm mentioning only System Settings for these because in our config file documentation, we tell people it's best to create the files by exporting them out of System Settings. Please let me know if this is not enough.

stian-sigvartsen replied:
Sounds good. That should be enough then.

stian-sigvartsen commented on discover/deployment/articles-dxp/03-configuring-liferay/04-securing-liferay/01-saml/03-configuring-saml.markdown:
```diff
@@ -0,0 +1,53 @@
+
+Note that there is also a system wide configuration, represented by the
+`SamlConfiguration` metatype. This uses the 
+[Configuration API](/develop/tutorials/-/knowledge_base/7-0/configuration)
```
It's actually the other way around, SamlProviderConfiguration is the one which uses Configuration API in order to read configuration at a company scope. The system scope only SamlConfiguration is handled by Configuration Admin natively.

sez11a replied:
Are you sure? This came directly from your text in the ticket: https://issues.liferay.com/browse/LRDOCS-3683

stian-sigvartsen commented on discover/deployment/articles-dxp/03-configuring-liferay/04-securing-liferay/01-saml/02-setting-up-service-provider.markdown:
```diff
@@ -0,0 +1,279 @@
+    - Entity ID: *liferaysamlidp*
+	- Clock Skew
+	- Force Authn
+    - Metadata URL: http://localhost:8080/c/portal/saml/metadata (test this URL
```
Maybe replace "localhost" with "[IdP host name]"

stian-sigvartsen reviewed (changes requested):
I think this reads really well overall. I've added inline comments. Also it looks like I am no longer able to specify configuration via files with .config extension. Now the extension must be .cfg from the looks of it. Maybe verify this with the config admin team?

stian-sigvartsen commented:
@sez11a As discussed I'll be out for the rest of the week. If you would like a second round of SME review then hopefully @csierra or @topolik can pick it up, but I think it will be good with these small tweaks.

sez11a committed:
LRDOCS-3817 Wordsmithing.

codyhoag committed:
Formatting for Greedy/Reluctant Blade sample

xbrianlee commented:
@stian-sigvartsen I was able to specify configurations using .config files, but not .cfg files (startup errors). Users should be able to use either extension as with configurations from other components. Isn't this a bug isolated to SAML?

csierra commented:
hey @xbrianlee, 
what startup errors do you get? It is weird because SAML module uses the same Configuration Admin APIs as the rest of the modules.

xbrianlee commented:
@csierra I modified the configurations that require integer values in `com.liferay.saml.runtime.configuration.SamlConfiguration.cfg` and got errors like the following:
```
17:14:59,135 ERROR [fileinstall-/Users/brianlee/Git/test-private7/7.0.x-private-bundles/osgi/modules][com_liferay_saml_opensaml_integration:97] [com.liferay.saml.opensaml.integration.internal.credential.DLKeyStoreManagerImpl(462)] The activate method has thrown an exception
java.lang.RuntimeException: Unable to create snapshot class for interface com.liferay.saml.runtime.configuration.SamlConfiguration
	at com.liferay.portal.configuration.metatype.bnd.util.ConfigurableUtil._createConfigurableSnapshot(ConfigurableUtil.java:77)
	at com.liferay.portal.configuration.metatype.bnd.util.ConfigurableUtil.createConfigurable(ConfigurableUtil.java:51)
	at com.liferay.saml.opensaml.integration.internal.credential.BaseKeyStoreManagerImpl.updateConfigurations(BaseKeyStoreManagerImpl.java:60)
	at com.liferay.saml.opensaml.integration.internal.credential.DLKeyStoreManagerImpl.activate(DLKeyStoreManagerImpl.java:138)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.felix.scr.impl.inject.BaseMethod.invokeMethod(BaseMethod.java:224)
	at org.apache.felix.scr.impl.inject.BaseMethod.access$500(BaseMethod.java:39)
	at org.apache.felix.scr.impl.inject.BaseMethod$Resolved.invoke(BaseMethod.java:617)
	at org.apache.felix.scr.impl.inject.BaseMethod.invoke(BaseMethod.java:501)
	at org.apache.felix.scr.impl.inject.ActivateMethod.invoke(ActivateMethod.java:302)
	at org.apache.felix.scr.impl.inject.ActivateMethod.invoke(ActivateMethod.java:294)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.createImplementationObject(SingleComponentManager.java:297)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.createComponent(SingleComponentManager.java:108)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.getService(SingleComponentManager.java:906)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.getServiceInternal(SingleComponentManager.java:879)
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.activateInternal(AbstractComponentManager.java:748)
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.enableInternal(AbstractComponentManager.java:674)
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.enable(AbstractComponentManager.java:429)
	at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.enableComponents(ConfigurableComponentHolder.java:657)
	at org.apache.felix.scr.impl.BundleComponentActivator.initialEnable(BundleComponentActivator.java:341)
	at org.apache.felix.scr.impl.Activator.loadComponents(Activator.java:403)
	at org.apache.felix.scr.impl.Activator.access$200(Activator.java:54)
	at org.apache.felix.scr.impl.Activator$ScrExtension.start(Activator.java:278)
	at org.apache.felix.utils.extender.AbstractExtender.createExtension(AbstractExtender.java:259)
	at org.apache.felix.utils.extender.AbstractExtender.modifiedBundle(AbstractExtender.java:232)
	at org.osgi.util.tracker.BundleTracker$Tracked.customizerModified(BundleTracker.java:482)
	at org.osgi.util.tracker.BundleTracker$Tracked.customizerModified(BundleTracker.java:1)
	at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:232)
	at org.osgi.util.tracker.BundleTracker$Tracked.bundleChanged(BundleTracker.java:444)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:905)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148)
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEventPrivileged(EquinoxEventPublisher.java:165)
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:75)
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:67)
	at org.eclipse.osgi.internal.framework.EquinoxContainerAdaptor.publishModuleEvent(EquinoxContainerAdaptor.java:102)
	at org.eclipse.osgi.container.Module.publishEvent(Module.java:461)
	at org.eclipse.osgi.container.Module.start(Module.java:452)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.start(EquinoxBundle.java:402)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.startBundle(DirectoryWatcher.java:1258)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.startBundles(DirectoryWatcher.java:1230)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.doProcess(DirectoryWatcher.java:512)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.process(DirectoryWatcher.java:361)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.run(DirectoryWatcher.java:312)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.liferay.portal.configuration.metatype.bnd.util.ConfigurableUtil._generateSnapshotClassData(ConfigurableUtil.java:179)
	at com.liferay.portal.configuration.metatype.bnd.util.ConfigurableUtil._createConfigurableSnapshot(ConfigurableUtil.java:64)
	... 46 more
Caused by: java.lang.IllegalArgumentException: No conversion to class java.lang.Integer from class java.lang.String value "61"
	at aQute.bnd.annotation.metatype.Configurable$ConfigurableHandler.convert(Configurable.java:194)
	at aQute.bnd.annotation.metatype.Configurable$ConfigurableHandler.invoke(Configurable.java:94)
	at com.sun.proxy.$Proxy21.getIdpSsoSessionCheckInterval(Unknown Source)
	... 52 more
```
The same file with `.config` extension didn't cause any errors.

csierra commented:
@xbrianlee 
I guess that's because the other format does not support integer values, only string values, even if you use numbers. I can't remember now if that format allows for you to _qualify_ the value as integer. 
So this is not a bug in SAML. It would fail the same way in any modules that expect an integer type in the configuration.

codyhoag committed:
LRDOCS-4015 NPM Samples intro

Commit:
LRDOCS-4014

xbrianlee commented:
@csierra You're right! I just tried with a different configuration and ran into startup errors using `.cfg `extension. I guess `.cfg` unofficially works with configurations that don't contain integer values.

@stian-sigvartsen If you're still unable to specific configurations using the `.config` extension, could you please elaborate on what you're seeing?

codyhoag committed:
LRDOCS-4015 Add sample READMEs drafts to LDN

codyhoag committed:
LRDOCS-4015 Add sample npm draft images

codyhoag committed:
LRDOCS-4015 Wrap 80 col, format, and configure image paths

rbohl committed:
LRDOCS-3977 address review and add info about -default

codyhoag committed:
LRDOCS-4015 Wordsmithing

mwilliams2014 committed:
LRDOCS-3748 wordsmithing

mwilliams2014 committed:
Merge branch 'LRDOCS-3748-lexicon-tut-refactor' into LRDOCS-3748-lexicon-tut-7.0.x

rbohl committed:
LRDOCS-3977 improvements based on review. more coming.

mwilliams2014 committed:
LRDOCS-3748 wordsmithing and screenshots

codyhoag committed:
LRDOCS-4015 Add additional info for npm samples

mwilliams2014 committed:
LRDOCS-3598 reorganizes upgrading themes tutorials

codyhoag committed:
Fix fragment path suggested by community member

Commit:
LRDOCS-4014 Explain why JARs are excluded

Commit:
LRDOCS-4014

codyhoag committed:
Document new check-links functionality and update branch info

codyhoag committed:
LRDOCS-4020 blade deploy only works for JARs

sez11a committed:
LRDOCS-3817 Wordsmithing.

stian-sigvartsen commented:
@xbrianlee Actually that appears to work. I was not escaping the "=" character properly. There were no exception in logs for this mistake so it was not obvious that was the issue

Commit:
Merge branch '7.0.x' of https://github.com/liferay/liferay-docs into 7.0.x

sez11a committed:
Merge pull request #2570 from codyhoag/gradle-ref-docs

Weekly Gradle plugin ref docs

rbohl committed:
LRDOCS-3977 factory configuration improvements based on review

sez11a committed:
Merge pull request #2572 from codyhoag/greedy-policy-formatting

Greedy Policy Formatting Fix

sez11a committed:
Merge branch '7.0.x' of github.com:sez11a/liferay-docs into 7.0.x

rbohl committed:
LRDOCS-3817 merge pr-2569 from sez11a

Commit:
added information on embedding templates within templates

rbohl committed:
LRDOCS-3817 explain workflow in introduction

sez11a committed:
LRDOCS-3951 Wordsmithing.

rbohl committed:
LRDOCS-3817 moved explanation of startWorkflowInstance to Guestbook article

codyhoag committed:
LRDOCS-3996 Begin Staging Triage testing doc

codyhoag committed:
LRDOCS-4027 Rename "Sample Modules" to "Sample Projects"

codyhoag committed:
LRDOCS-4027 folder rename: modules -> projects

codyhoag committed:
LRDOCS-4027 folder rename: modules -> projects

codyhoag committed:
LRDOCS-4027 Update intro

codyhoag committed:
LRDOCS-4027 Using modules here makes more sense

codyhoag committed:
LRDOCS-4027 Delete sample-modules folder

codyhoag committed:
LRDOCS-4027 Delete sample-modules folder

# Conflicts:
#	develop/reference/articles/13-sample-modules/apps/01-greedy-policy-option-portlet.markdown

codyhoag committed:
LRDOCS-4026 Current toolbar config only applicable to AlloyEditor

codyhoag committed:
LRDOCS-4027 Copy over npm docs

codyhoag committed:
LRDOCS-4027 Delete old npm docs too to accomodate for renamed folder

codyhoag committed:
LRDOCS-4015 Ivan's suggestions

codyhoag committed:
LRDOCS-4015 Add sample location section and important version info

codyhoag committed:
LRDOCS-4015 number headers

codyhoag committed:
Merge branch '7.0.x' into npm-sample-docs

codyhoag committed:
LRDOCS-4015 Wordsmithing

Commit:
updated instructions for embedding template and added image

codyhoag committed:
LRDOCS-4030 Remove licensing and contributing section from DXP version of Ext doc

codyhoag committed:
LRDOCS-4030 Number header

mwilliams2014 committed:
Merge branch 'LRDOCS-3564-kb-staging-note' into LRDOCS-3564-staging-note-7.0.x

mwilliams2014 committed:
LRDOCS-3564 updated version

codyhoag committed:
LRDOCS-3996 Write intro and enter remaining table info for web content entities

codyhoag committed:
LRDOCS-3996 Table formatting

Commit:
LRDOCS-4031 Describe DBStore's file size limitation and File System Store alternative.

Commit:
LRDOCS-4031 Clarify that DBStore stores files as blobs.

sez11a committed:
LRDOCS-3748 Wordsmithing.

sez11a committed:
Merge branch 'pr-2576' into 7.0.x

sez11a committed:
Merge branch 'LRDOCS-3598-upgrading-themes-reorg-7.0.x' of github.com:mwilliams2014/liferay-docs into 7.0.x

rbohl committed:
LRDOCS-3977 factory configuration revisions

sez11a committed:
Merge pull request #2579 from codyhoag/resources-typo

Fix fragment path suggested by community member

sez11a committed:
Merge pull request #2581 from codyhoag/liferay-docs-guideline-updates

Document new check-links functionality and update branch info | 7.0.x

sez11a committed:
Merge pull request #2583 from codyhoag/blade-deploy

LRDOCS-4020 "blade deploy" only for JARs

sez11a committed:
Merge pull request #2584 from jhinkey/bypassing-jar-exclusion

LRDOCS-4014 Understanding excluded JARs

sez11a committed:
Header IDs

sez11a committed:
Merge pull request #2593 from codyhoag/ext-update

LRDOCS-4030 Incorrect licensing info removed from DXP doc

mwilliams2014 committed:
Merge branch 'patch-6' of https://github.com/chrislui/liferay-docs into LRDOCS-3786-rss-portlet-note

mwilliams2014 committed:
LRDOCS-3786 wordsmithing

codyhoag committed:
Refer to Elasticsearch correctly

mwilliams2014 committed:
LRDOCS-4036 SF

rbohl committed:
LRDOCS-4034 fix broken link

phelgren committed:
Add specific reference to ViewPortletProvider class

Clarifies where to put the property since the @Component directive is in both Java files

codyhoag committed:
Fix various links using check-links tool

codyhoag committed:
These links are invalid and this section seems out of place in the learning path (no other articles have this section). Because of this, I removed it all together

codyhoag committed:
Weekly Gradle plugin reference doc updates

codyhoag committed:
LRDOCS-3996 Wordsmithing

codyhoag committed:
LRDOCS-3996 Final wordsmithing

codyhoag committed:
LRDOCS-4041 Provide more explanation for tutorial to avoid confusion if users look at linked example

rbohl committed:
LRDOCS-3977 final edits

sez11a committed:
LRDOCS-3817 Wordsmithing.

sez11a committed:
Merge branch 'pr-2590' into 7.0.x

sez11a committed:
Merge branch '7.0.x' of github.com:sez11a/liferay-docs into 7.0.x

sez11a committed:
Header IDs.

mwilliams2014 committed:
LRDOCS-3482 added markup to workflow section

codyhoag committed:
Breaking change update

stiemannkj1 committed:
LRDOCS-4203 Liferay Faces branch versions and links are out of date in "Liferay Faces Version Scheme"

codyhoag committed:
LRDOCS-4037 Print article path for check-links

stiemannkj1 committed:
LRDOCS-4042 Improve artifact info in "Liferay Faces Version Scheme"

sez11a committed:
Merge pull request #2612 from mwilliams2014/LRDOCS-3842-workflow-styles

LRDOCS-3482 added markup to workflow section

sez11a committed:
Formatting

codyhoag committed:
Merge branch 'fix-liferay-faces-version-docs-LRDOCS-4042-4043-7.0.x' of https://github.com/stiemannkj1/liferay-docs into faces-version-scheme-7.0.x

codyhoag committed:
LRDOCS-4042 Wordsmithing

codyhoag committed:
Breaking change - update prefix for tags

codyhoag committed:
SF plugin version updates

sez11a committed:
Merge pull request #2589 from codyhoag/sample-modules-rename

LRDOCS-4027 Sample Modules Rename

sez11a committed:
Merge pull request #2591 from codyhoag/modifying-editor-updates

LRDOCS-4026 WYSIWYG toolbar config only applicable to AlloyEditor

sez11a committed:
Merge pull request #2594 from mwilliams2014/LRDOCS-3564-staging-note-7.0.x

LRDOCS-3564 staging note 7.0.x

sez11a committed:
Merge pull request #2595 from stephenkostas/LRDOCS-1510-embed-templates

LRDOCS-1510 Add instructions on embedding templates

sez11a committed:
Merge pull request #2596 from jhinkey/dbstore-limitation-LRDOCS-4031

LRDOCS-4031 Describe DBStore's file size limitation and alternative

sez11a committed:
Merge pull request #2599 from mwilliams2014/LRDOCS-3786-rss-portlet-note

LRDOCS-3786 RSS portlet note

sez11a committed:
Merge pull request #2600 from codyhoag/elasticsearch-typos

Typo fixes

sez11a committed:
Merge pull request #2601 from mwilliams2014/LRDOCS-4036-note-fix-7.0.x

LRDOCS-4036 SF

sez11a committed:
Merge pull request #2603 from rbohl/LRDOCS-4034-broken-link-shield

LRDOCS-4034, fix link

sez11a committed:
Merge pull request #2605 from codyhoag/fix-links

Fixed various links using check-links tool

sez11a committed:
Merge pull request #2607 from codyhoag/gradle-ref-docs

Weekly Gradle plugin reference doc updates

codyhoag committed:
Merge branch '7.0.x' into npm-sample-docs

# Conflicts:
#	develop/reference/articles/13-sample-modules/apps/02-shared-language-keys.markdown
#	develop/reference/articles/13-sample-modules/apps/03-simulation-panel-app.markdown
#	develop/reference/articles/13-sample-modules/apps/04-spring-mvc-portlet.markdown
#	develop/reference/articles/13-sample-projects/apps/02-shared-language-keys.markdown
#	develop/reference/articles/13-sample-projects/apps/03-shared-language-keys.markdown
#	develop/reference/articles/13-sample-projects/apps/03-simulation-panel-app.markdown
#	develop/reference/articles/13-sample-projects/apps/04-simulation-panel-app.markdown
#	develop/reference/articles/13-sample-projects/apps/04-spring-mvc-portlet.markdown
#	develop/reference/articles/13-sample-projects/apps/05-spring-mvc-portlet.markdown

sez11a committed:
Merge pull request #2592 from codyhoag/npm-sample-docs

LRDOCS-4015 npm Sample Docs

sez11a committed:
LRDOCS-3996 Wordsmithing.

sez11a committed:
Merge branch 'pr-2609' into 7.0.x

sez11a committed:
Merge pull request #2610 from codyhoag/user-personal-bar-updates

LRDOCS-4041 User personal bar updates

sez11a committed:
LRDOCS-3977 Wordsmithing.

sez11a committed:
Merge branch 'pr-2611' into 7.0.x

sez11a committed:
Merge branch '7.0.x' of github.com:sez11a/liferay-docs into 7.0.x

sez11a committed:
Merge pull request #2613 from codyhoag/breaking-change-update

Breaking change update

sez11a committed:
Merge pull request #2614 from codyhoag/check-links-parent-folder

LRDOCS-4037 Print article path for check-links | 7.0.x

sez11a committed:
Merge pull request #2617 from codyhoag/faces-version-scheme-7.0.x

LRDOCS-4042 Update Faces version scheme | 7.0.x

sez11a committed:
Merge branch '7.0.x' into LRDOCS-3683-SAML-update

sez11a committed:
LRDOCS-3683 Fixes from SME Review

sez11a committed:
Header IDs

sez11a committed:
Somehow this was changed to 7.1

stian-sigvartsen commented:
I’m very sorry but I made a mistake on the ticket. SamlConfiguration is NOT read via Configuration API. It is provided to components via standard methods (activation method) and query to Config Admin API.  I’ve removed the incorrect sentence from the ticket.


> On 27 Oct 2017, at 20:29, Rich Sezov <notifications@github.com> wrote:
> 
> @sez11a commented on this pull request.
> 
> In discover/deployment/articles-dxp/03-configuring-liferay/04-securing-liferay/01-saml/03-configuring-saml.markdown <https://github.com/stian-sigvartsen/liferay-docs/pull/1#discussion_r147497690>:
> 
> > +	- `saml.entity.id`
> +	- `saml.sp.ldap.import.enabled`
> +	- `saml.role`
> +	- `saml.idp.session.maximum.age`
> +    - `saml.idp.session.timeout`
> +    - `saml.sp.sign.authn.request`
> +    - `saml.sign.metadata`
> +    - `saml.ssl.required`
> +    - `saml.idp.metadata.name.id.attribute`
> +
> +The SAML Admin UI remains the place for creating the company-scoped
> +configuration instances.
> +
> +Note that there is also a system wide configuration, represented by the
> +`SamlConfiguration` metatype. This uses the 
> +[Configuration API](/develop/tutorials/-/knowledge_base/7-0/configuration)
> Are you sure? This came directly from your text in the ticket: https://issues.liferay.com/browse/LRDOCS-3683 <https://issues.liferay.com/browse/LRDOCS-3683>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/stian-sigvartsen/liferay-docs/pull/1#discussion_r147497690>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AL48X5AZOBnK2hnQ7E9pbeu3eaxLmGGOks5swi8YgaJpZM4Prf3r>.
>

sez11a committed:
Merge branch '7.0.x' into LRDOCS-3683-SAML-update

sez11a committed:
LRDOCS-3683 Last edit from SME Review.

sez11a commented:
Okay; fixed!
